sig-architecture/api-review-process.md

```diff
@@ -0,0 +1,257 @@
+
+* A KEP has been created if introducing:
+
+    * Any new resource type
+
+    * Any new version of a stable API
+
+    * Any new functionality added to a stable API as defined by SIG Architecture and the API Reviewers
+
+## What APIs need to be reviewed?
+
+* What are the kind of reviews?
+
+    * **mandatory**: The changer is expected to submit the changes to API review, and the change is blocked on the API reviewer approval 
+
+    * **voluntary**: Changer can request a review, or a 3rd party can nominate the API for review.  The review comments are recommendations.
+
+* Which projects are in scope?
+
+    * github.com/kubernetes/kubernetes, and any kubernetes-* org projects it depends on are required to be reviewed.   (mandatory)
+
+    * Any kubernetes-style API that uses a *.k8s.io or *.kubernetes.io name, e.g. "storage.k8s.io" (mandatory)
+
+    * "Critical" or other “highly-integrated” APIs, such as our extension points in the node, apiserver, and controllers as defined by SIG Architecture [TBD].  (mandatory)
+
+    * any other github projects in the kubernetes-* orgs (informational, for now - may make mandatory in future)
+
+    * Any projects that produce APIs (including CRDs) that are intended to work with kubectl and/or kube-apiserver. (informational - intent is to ensure consistent user experience across the Kubernetes ecosystem)
+
+* All API implementations (including alpha versions) that are part of kubernetes must be reviewed, including CRDs, so user experience across the Kubernetes ecosystem is consistent.
+
+* What parts of a PR are "API changes"?
+
+    * "Resource APIs" include the versioned data definition (pkg/apis/*/v*/types.go or OpenAPI for CRDs), validation (pkg/apis/*/validation.go or OpenAPI for CRDs).
+
+    * Configuration files, flags, and command line arguments are all part of our user and script facing APIs and must be reviewed.
+
+    * Compiled-in APIs of the kube-apiserver
+
+    * Webhooks request/response formats in kube-apiserver
+
+    * HTTP APIs in kubelet
+
+    * plugins which are not covered by some other standards effort (e.g. CSI and CNI APIs would be deferred to those standards bodies)
+
+## End-states of Reviews
+
+The API review process can result in multiple outcomes depending on the content of the change. For example, a new API could be approved for being builtin, or rejected in favor of out-of-tree development. An API that is reviewed informationally, results in either suggestions for changes, or approval as-is. 
+
+## Information required from the submitter
+
+As much as possible, we will automate the detection of PRs that require API reviews.  Significant changes should be reviewed BEFORE they reach the PR stage, but this is the backstop for anything that gets missed.  Such automation may miss cases, so any PR can be flagged as "needs API review", which triggers this process.
+
+To begin the process:
```

Comment:
some overlap with the "Process mechanics" section below, might collapse these as a follow up